workers/cpp_physics/physics.cc

```diff
@@ -73,15 +75,37 @@ int main(int argc, char** argv) {
   worker::View view{ComponentRegistry{}};
   bool is_connected = true;
 
+  using AssignParititionCommand = improbable::restricted::Worker::Commands::AssignPartition;
+  // In real code, we would probably want to retry here.
+  view.OnCommandResponse<AssignParititionCommand>(
+      [&](const worker::CommandResponseOp<AssignParititionCommand>& op) {
+        if (op.StatusCode == worker::StatusCode::kSuccess) {
+          connection.SendLogMessage(worker::LogLevel::kInfo, "Physics",
+                                    "Successfully assigned partition.");
+        } else {
+          connection.SendLogMessage(worker::LogLevel::kError, "Physics",
+                                    "Failed to assign partition: error code : " +
+                                        std::to_string(static_cast<std::uint8_t>(op.StatusCode)) +
+                                        " message: " + op.Message);
+        }
+      });
+
+  connection.SendCommandRequest<improbable::restricted::Worker::Commands::AssignPartition>(
+      connection.GetWorkerEntityId(), {physicsPartitionId},
+      /* default timeout */ {});
+
   view.OnDisconnect([&](const worker::DisconnectOp& op) {
     std::cerr << "[disconnect] " << op.Reason << std::endl;
     is_connected = false;
   });
   view.OnCommandRequest<sample::Login::Commands::TakeControl>(
       [&](const worker::CommandRequestOp<sample::Login::Commands::TakeControl>& op) {
         connection.SendLogMessage(worker::LogLevel::kInfo, "Physics",
-                                  "Assigning ClientData component to worker with ID " +
+                                  "Assigning the client partition to worker with ID " +
                                       std::to_string(op.CallerWorkerEntityId));
+        connection.SendCommandRequest<AssignParititionCommand>(op.CallerWorkerEntityId,
```

Comment:
I might have just missed this, but what gives the physics worker permission to assign partitions to itself and other workers?

Reply:
The ability to send system commands, as per the `default_launch.json`. There are no other permission checks in place.